cmd/cosmos-sdk-cli/cmd/init.go

```diff
@@ -0,0 +1,132 @@
+package cmd
+
+import (
+	"fmt"
+	"go/build"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/spf13/cobra"
+	"path/filepath"
+)
+
+func init() {
+	initCmd.Flags().StringVarP(&remoteProjectPath, "project-path", "p", "", "Remote project path. eg: github.com/your_user_name/project_name")
+	rootCmd.AddCommand(initCmd)
+}
+
+func resolveProjectPath(remoteProjectPath string) string {
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		gopath = build.Default.GOPATH
+		// Use $HOME/go
+	}
+	return gopath + string(os.PathSeparator) + "src" + string(os.PathSeparator) + remoteProjectPath
+}
+
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+var remoteBasecoinPath = "github.com/cosmos/cosmos-sdk/examples/basecoin"
+
+func copyBasecoinTemplate(projectName string, projectPath string, remoteProjectPath string) {
+	basecoinProjectPath := resolveProjectPath(remoteBasecoinPath)
+	filepath.Walk(basecoinProjectPath, func(path string, f os.FileInfo, err error) error {
+		if !f.IsDir() {
+			data, err := ioutil.ReadFile(path)
+			check(err)
```

Comment:
This is unconventional to our codebase - the check function should removed and this line should be replaced with 
```golang 
if err != nil {
    return err
}
```